components/server/src/workspace/workspace-starter.ts

```diff
@@ -1294,47 +1285,17 @@ export class WorkspaceStarter {
         instance: WorkspaceInstance,
         lastValidWorkspaceInstanceId: string,
         ideConfig: IdeServiceApi.ResolveWorkspaceConfigResponse,
-        userEnvVars: UserEnvVarValue[],
-        projectEnvVars: ProjectEnvVar[],
+        envVars: ResolvedEnvVars,
     ): Promise<StartWorkspaceSpec> {
         const context = workspace.context;
 
-        let allEnvVars: EnvVarWithValue[] = [];
-        if (userEnvVars.length > 0) {
-            if (CommitContext.is(context)) {
-                // this is a commit context, thus we can filter the env vars
-                allEnvVars = allEnvVars.concat(
-                    UserEnvVar.filter(userEnvVars, context.repository.owner, context.repository.name),
-                );
-            } else {
-                allEnvVars = allEnvVars.concat(userEnvVars);
-            }
-        }
-        if (projectEnvVars.length > 0) {
-            // Instead of using an access guard for Project environment variables, we let Project owners decide whether
-            // a variable should be:
-            //   - exposed in all workspaces (even for non-Project members when the repository is public), or
-            //   - censored from all workspaces (even for Project members)
-            let availablePrjEnvVars = projectEnvVars;
-            if (workspace.type !== "prebuild") {
-                availablePrjEnvVars = projectEnvVars.filter((variable) => !variable.censored);
-            }
-            const withValues = await this.projectDB.getProjectEnvironmentVariableValues(availablePrjEnvVars);
-            allEnvVars = allEnvVars.concat(withValues);
-        }
-        if (WithEnvvarsContext.is(context)) {
-            allEnvVars = allEnvVars.concat(context.envvars);
-        }
-
-        const envvars: EnvironmentVariable[] = [];
-
         // TODO(cw): for the time being we're still pushing the env vars as we did before.
```

Comment:
@csweichel here an idea was that we don't pass it anymore, but supervisor fetches on start after getting the token?

I noticed that ws-manager has a filter for workspace env vars: https://github.com/gitpod-io/gitpod/blob/dda547c7a2803b36457998a1440bc9485e9be439/components/ws-manager-mk2/controllers/create.go#L605-L623

Should not we rather filter them here? that this behaviour is consistent with `gp env`?

Reply:
We have now 2 kind of env vars on spec, one configured by user for a workspace (user, project, context), and then system env vars (which are not filtered).

Let's move `GITPOD_WORKSPACE_CONTEXT` and so on to system env vars, and then filter the rest in `EnvVarService`?

Reply:
It is complicated because of protected secrets. If supervisor would fetch env vars instead it would be possible to simplify since we don't leave env vars on k8s spec then. I will leave it out for another PR.